src/mdoc/model/DeviceResponse.ts

```diff
@@ -81,20 +83,81 @@ export class DeviceResponse {
   }
 
   /**
-   * Set the session transcript data to use for the device response. This is arbitrary and should match
-   * the session transcript as it will be calculated by the verifier.
-   * We expect a buffer of bytes, as defined in as defined in ISO/IEC 18013-7 in both Annex A (Web API) and Annex B (OID4VP)
+   * Set the session transcript data to use for the device response. This is arbitrary and should match the session transcript as it will be calculated by the verifier.
+   * The transcript must be a CBOR encoded DataItem of an array with 3 elements, there is no further requirement.
+   * `usingSessionTranscriptBytes(cborEncode(DataItem.fromData([a,b,c])))` where `a`, `b` and `c` can be 
+    )
+   * It is preferable to use `usingSessionTranscriptBytesForOID4VP` or `usingSessionTranscriptBytesForWebAPI` when possible.
    * @param {Buffer} sessionTranscriptBytes - The sessionTranscriptBytes data to use in the session transcript.
    * @returns {DeviceResponse}
    */
   public usingSessionTranscriptBytes(sessionTranscriptBytes: Buffer): DeviceResponse {
     if (this.sessionTranscriptBytes) {
-      throw new Error('A session transcript has already been set, either with .usingHandover or .usingSessionTranscriptBytes');
+      throw new Error(
+        'A session transcript has already been set, with one of the .usingSessionTranscriptBytesXXX methods',
+      );
     }
     this.sessionTranscriptBytes = sessionTranscriptBytes;
     return this;
   }
 
+  /**
+   * Set the session transcript data to use for the device response as defined in ISO/IEC 18013-7 in Annex B (OID4VP).
+   * This should match the session transcript as it will be calculated by the verifier.
+   * @param {string} mdocGeneratedNonce -
+   * @param {string} clientId -
+   * @param {string} responseUri -
+   * @param {string} verifierGeneratedNonce -
+   * @returns {DeviceResponse}
+   */
+  public usingSessionTranscriptBytesForOID4VP(
+    mdocGeneratedNonce: string,
+    clientId: string,
+    responseUri: string,
+    verifierGeneratedNonce: string,
+  ): DeviceResponse {
+    this.usingSessionTranscriptBytes(
```

Comment:
check if `this.sessionTranscriptBytes` is already set

```
if (this.sessionTranscriptBytes) {
      throw new Error(
        'A session transcript has already been set, either with .usingSessionTranscriptBytesForOID4VP or .usingSessionTranscriptBytesForWebAPI',
      );
    }
```

Reply:
`usingSessionTranscriptBytes` already does so, the check in `usingSessionTranscriptForWebAPI` was a leftover